### DISCUSSION TITLE
Oxanium Font is always vertically aligned to the top, or is Avalonia not rendering it correctly?

### DISCUSSION THREAD

**Discussion by AngryCarrot789** [AvaloniaUI/Avalonia#15262] [Q&A]
Hi! Currently porting my WPF app to Avalonia, and the Oxanium font it uses doesn't seem to get rendered correctly.
Here's what I see in avalonia (The image isn't clipped; it's the entire client area. One pixel above is the titlebar):
![image](https://github.com/AvaloniaUI/Avalonia/assets/32274404/06181194-5935-430c-854d-1eb745c3f2d7)

And then this is what I see in WPF (Same as above, one pixel above is the titlebar area):
![image](https://github.com/AvaloniaUI/Avalonia/assets/32274404/61150dea-77ba-4b54-b5a6-09a993ebc230)

The menu items' text is offset towards the top, the tabs' text is slightly offset to the top, and the text in the bottom right corner is massively offset towards the top. I can fix all of this by manually adding a margin of `0,5,0,0` to most things but that's not really convenient...

This is my XAML code for the bottom-right text block, which is the exact same for WPF and Avalonia:
```
<TextBlock x:Name="PART_CaretText" DockPanel.Dock="Right" Margin="0,0,5,0" VerticalAlignment="Center"/>
```

And this is how I'm loading the font (it's placed in the Application.Resources at the very top, so it's always accessible anywhere):
```
<FontFamily x:Key="FontOxanium">/Resources/Fonts/Oxanium/#Oxanium</FontFamily>
```

Does anyone know what is causing this issue? Thanks!

**Comment by AngryCarrot789** [comment]
I tested out a TextBlock as the header for a menu item, where the background is set to orange and the menu item's background is blue. 

This is what `Segoe UI` looks like
![image](https://github.com/AvaloniaUI/Avalonia/assets/32274404/e3be03c1-5409-47a5-b97b-39ac1da37d2f)

This is Oxanium:
![image](https://github.com/AvaloniaUI/Avalonia/assets/32274404/a06758b4-1ad2-4d8f-8ef8-d92581f09794)

**Comment by Gillibald** [comment]
What Avalonia version are you using?

**Reply by AngryCarrot789** [reply]
Version 11.0.10 (for Avalonia, Avalonia.Desktop, Fonts.Inter, etc.). I also did the same tests for 11.0.0

I don't know if this has anything to do with it, but the text measurements seem to rely on the `TextMetrics` struct, and I noticed that `LineHeight` includes `LineGap`, and that line gap (after scaling) for the Oxanium font was 3 pixels which seems to make up that empty space I'm getting

**Comment by AngryCarrot789** [comment]
This is apparently a known issue in other frameworks like Java Swing with fonts like Consolas:
https://bugs.openjdk.org/browse/JDK-8203482?page=com.atlassian.jira.plugin.system.issuetabpanels%3Achangehistory-tabpanel

**Comment by timunie** [comment]
In case you have an idea how to correct it let us know your thoughts.

**Reply by AngryCarrot789** [reply]
From what I've read at least for windows, FreeType has 2 metrics called usWinAscent and usWinDescent can be used for determining the top and bottom of the text, based on this comment: https://youtrack.jetbrains.com/issue/IDEA-225763/the-baseline-of-the-consolas-font-is-incorrect#focus=Comments-27-4623283.0-0. Not sure if it's a problem with Avalonia, Harfbuzz or another lib

**Reply by Gillibald** [reply]
https://github.com/toptensoftware/RichTextKit/pull/67#issuecomment-1980973137

**Reply by AngryCarrot789** [reply]
So I guess this means that avalonia requires a platform-specific font metric calculator or something along those lines?

**Comment by AngryCarrot789** [comment]
I opened a PR which fixes this issue

https://github.com/AvaloniaUI/Avalonia/pull/15344

**Comment by AngryCarrot789** [comment]
It seems like this issue is back with version 11.3.5?

This is 11.3.4
<img width="295" height="90" alt="image" src="https://github.com/user-attachments/assets/67886aeb-82b9-42ce-b5f0-6920c0b987ee" />

Then this is 11.3.5
<img width="295" height="86" alt="image" src="https://github.com/user-attachments/assets/b7064861-97f1-4c02-99d7-09a527ccce35" />

This is my xaml code:
```xaml
<TextBox Grid.Column="2" Grid.Row="0" FontFamily="Consolas"
         Background="{DynamicResource ABrush.MemEngine.MainView.AdditionOptions.TextBox.Background}"
         BorderBrush="{DynamicResource ABrush.MemEngine.MainView.AdditionOptions.TextBox.Border}"
         Foreground="{DynamicResource ABrush.MemEngine.MainView.AdditionOptions.TextBox.Foreground}"
         x:Name="PART_ScanOption_StartAddress" Padding="4,1"
         VerticalAlignment="Stretch" VerticalContentAlignment="Center"
         HorizontalAlignment="Stretch" HorizontalContentAlignment="Stretch"
         ScrollViewer.VerticalScrollBarVisibility="Disabled"
         ScrollViewer.HorizontalScrollBarVisibility="Hidden">
    <TextBox.InnerLeftContent>
        <TextBlock Text="0x" Opacity="0.7"/>
    </TextBox.InnerLeftContent>
</TextBox>
```
